src/Microsoft.ML.TimeSeries/SrCnnBatchAnomalyDetection.cs

```diff
@@ -0,0 +1,778 @@
+    /// The detect modes of SrCnn models.
+    /// </summary>
+    public enum SrCnnDetectMode
+    {
+        /// <summary>
+        /// In this mode, output (IsAnomaly, RawScore, Mag).
+        /// </summary>
+        AnomalyOnly = 0,
+
+        /// <summary>
+        /// In this mode, output (IsAnomaly, AnomalyScore, Mag, ExpectedValue, BoundaryUnit, UpperBoundary, LowerBoundary).
+        /// </summary>
+        AnomalyAndMargin = 1,
+
+        /// <summary>
+        /// In this mode, output (IsAnomaly, RawScore, Mag, ExpectedValue).
+        /// </summary>
+        AnomalyAndExpectedValue = 2
+    }
+
+    // TODO: SrCnn
+    internal sealed class SrCnnBatchAnomalyDetector : BatchDataViewMapperBase<double, SrCnnBatchAnomalyDetector.Batch>
+    {
+        private const int MinBatchSize = 12;
+        private const int AnomalyOnlyOutputLength = 3;
+        private const int AnomalyAndExpectedValueOutputLength = 4;
+        private const int AnomalyAndMarginOutputLength = 7;
+
+        private readonly int _batchSize;
+        private readonly string _inputColumnName;
+        private readonly int _outputLength;
+        private readonly SrCnnEntireModeler _modler;
+
+        private class Bindings : ColumnBindingsBase
+        {
+            private readonly VectorDataViewType _outputColumnType;
+            private readonly int _inputColumnIndex;
+
+            public Bindings(DataViewSchema input, string inputColumnName, string outputColumnName, VectorDataViewType outputColumnType)
+                : base(input, true, outputColumnName)
+            {
+                _outputColumnType = outputColumnType;
+                _inputColumnIndex = Input[inputColumnName].Index;
+            }
+
+            protected override DataViewType GetColumnTypeCore(int iinfo)
+            {
+                Contracts.Check(iinfo == 0);
+                return _outputColumnType;
+            }
+
+            // Get a predicate for the input columns.
+            public Func<int, bool> GetDependencies(Func<int, bool> predicate)
+            {
+                Contracts.AssertValue(predicate);
+
+                var active = new bool[Input.Count];
+                for (int col = 0; col < ColumnCount; col++)
+                {
+                    if (!predicate(col))
+                        continue;
+
+                    bool isSrc;
+                    int index = MapColumnIndex(out isSrc, col);
+                    if (isSrc)
+                        active[index] = true;
+                    else
+                        active[_inputColumnIndex] = true;
+                }
+
+                return col => 0 <= col && col < active.Length && active[col];
+            }
+        }
+
+        public SrCnnBatchAnomalyDetector(IHostEnvironment env, IDataView input, string inputColumnName, string outputColumnName, double threshold, int batchSize, double sensitivity, SrCnnDetectMode detectMode)
+            : base(env, "SrCnnBatchAnomalyDetector", input)
```

Comment:
>SrCnnBatchAnomalyDetector [](start = 25, length = 25)

Please use `nameof(SrCnnBatchAnomalyDetector)`.

Reply:
fixed,

---
In reply to: [426396967](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426396967) [](ancestors = 426396967)